source/extensions/http/header_validators/envoy_default/header_validator.cc

```diff
@@ -434,6 +434,49 @@ bool HeaderValidator::hasChunkedTransferEncoding(const HeaderString& value) {
   return false;
 }
 
+// For all (H/1, H/2 and H/3) protocols, trailers should only contain generic headers. As such a
+// common validation method can be used. For H/2, per
+// https://www.rfc-editor.org/rfc/rfc9113#section-8.1 trailers MUST NOT contain H/2 pseudo header
+// fields. For H/1 per https://www.rfc-editor.org/rfc/rfc9110#section-6.5 there are no specific
+// prohibitions.
```

Comment:
for H2 and H3 no pseudo headerrs.  if we disallow for H1 as well let's say there's no prohibitions but we disallow anyway?

Reply:
Clarified comment. I can also remove the more detailed explanation and just link to RFCs.